libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -375,13 +385,19 @@ export function OutQueueManager(
           clearTimeout(xhrTimeout);
           if (xhr.status >= 200 && xhr.status < 300) {
             onPostSuccess(numberToSend);
-          } else {
-            if (!shouldRetryForStatusCode(xhr.status)) {
-              LOG.error(`Status ${xhr.status}, will not retry.`);
-              removeEventsFromQueue(numberToSend);
-            }
-            executingQueue = false;
+            return;
+          }
+
+          if (xhr.status === 0) {
```

Comment:
Nit: can we use `else if` instead of the `return` and `if`? I think it's easier  to follow and makes it less likely we forget a `return` somewhere.